src/client/contextChannels.ts

```diff
@@ -183,6 +259,56 @@ export abstract class ChannelBase {
         }
         return listener;
     }
+
+    /**
+     * Event that is fired whenever a window changes joins this channel. This includes switching to/from the default
+     * channel.
+     *
+     * The event also includes which channel the window was in previously. The `channel` property within the
+     * event will always be this channel instance.
+     */
+    public async addEventListener(eventType: 'window-added', listener: (event: ChannelWindowAddedEvent) => void): Promise<void>;
+
+    /**
+     * Event that is fired whenever a window changes leaves this channel. This includes switching to/from the default
+     * channel.
+     *
+     * The event also includes which channel the window is being added to. The `previousChannel` property within the
+     * event will always be this channel instance.
+     */
+    public async addEventListener(eventType: 'window-removed', listener: (event: ChannelWindowRemovedEvent) => void): Promise<void>;
+
+    public async addEventListener<E extends FDC3ChannelEvent>(eventType: E['type'], listener: (event: E) => void): Promise<void> {
+        validateEnvironment();
+
+        const hasEventListenerBefore = channelEventEmitters[this.id].listenerCount(eventType) > 0;
+        channelEventEmitters[this.id].addListener(eventType, listener);
+
+        if (!hasEventListenerBefore) {
+            await tryServiceDispatch(APIFromClientTopic.CHANNEL_ADD_EVENT_LISTENER, {id: this.id, eventType});
+        }
+    }
+
+    public async removeEventListener(eventType: 'window-added', listener: (event: ChannelWindowAddedEvent) => void): Promise<void>;
+    public async removeEventListener(eventType: 'window-removed', listener: (event: ChannelWindowRemovedEvent) => void): Promise<void>;
+
+    /**
+     * Removes a listener previously added with {@link addEventListener}.
+     *
+     * Has no effect if `eventType` isn't a valid event, or `listener` isn't a callback registered against `eventType`.
+     *
+     * @param eventType The event being unsubscribed from
+     * @param listener The callback function to remove
+     */
+    public async removeEventListener<E extends FDC3ChannelEvent>(eventType: E['type'], listener: (event: E) => void): Promise<void> {
+        validateEnvironment();
+
+        channelEventEmitters[this.id].removeListener(eventType, listener);
```

Comment:
`channelEventEmitters` gets used globally here:

        eventHandler.registerEmitterProvider('channel', (channelId: ChannelId) => {
            return channelEventEmitters[channelId];
        });

so if it lived on the Channel, it would need to be public

As a secondary "ugh I don't want to change this", having the emitter live on the ConxtextChannel caused problems when the Channels are sent across the wire as part of FDC3RemoteExecution